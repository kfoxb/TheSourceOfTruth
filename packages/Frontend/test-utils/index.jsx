/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
