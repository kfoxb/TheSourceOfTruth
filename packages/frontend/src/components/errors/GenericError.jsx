import React from 'react';
import { withRouter } from 'react-router';
import PropTypes from 'prop-types';
import Dialog from '../Dialog';

function GenericError({ history }) {
  return (
    <Dialog
      dialogIsOpen
      title="System Error"
      content="Something went wrong. Please try again later."
      buttons={[
        {
          label: 'Back to Home',
          action: () => { history.push('/'); },
        },
      ]}
    />
  );
}

GenericError.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default withRouter(GenericError);
