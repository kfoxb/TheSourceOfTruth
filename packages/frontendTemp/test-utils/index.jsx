/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
