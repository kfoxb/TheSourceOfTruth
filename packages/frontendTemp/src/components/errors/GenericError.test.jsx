/* eslint-env jest */
import { createAssertWithPropsToMatchSnapshot } from '../../../test-utils';
import GenericError from './GenericError';

describe('GenericError', () => {
  const defaultProps = {
    history: {
      push: () => {},
    },
  };
  it('should render', () => {
    createAssertWithPropsToMatchSnapshot(GenericError, defaultProps)();
  });
});
