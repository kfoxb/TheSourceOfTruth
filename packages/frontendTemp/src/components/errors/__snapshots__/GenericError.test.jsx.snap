// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericError should render 1`] = `
<Route
  render={[Function]}
/>
`;
